### ISSUE TITLE
New Openmediavault Install - Error about Monit and Socket 2812 when hitting apply

### ISSUE THREAD

**Issue by vepicurean** [openmediavault/openmediavault#435]
Hi:

I installed Openmediavault 4 about a month ago. For the most part it worked OK (some small issues on a notification that came across every week, but it was a minor annoyance rather than a functional issue).  I know when I installed and tested the server and Openmediavault, things were working as they should.

Openmediavault and all the plug-ins are fully up-to-date.  All I've done is added a couple AFP and SMB shares and run the updates. Nothing else. Basic stuff. 

Suddenly, this week, most times (not overtime, most times) when I hit apply I get this strange error about Monit and socket 2812 comes up (the apply fails miserably). No where do I see any information on this issue that makes sense to me.  

I basically can't do anything with Openmediavault right now. Every time I try to change and apply something, the apply fails and I have to revert. 

How is this happening on a new install?

**Comment by votdev** [comment]
Which plugins do you have installed?

**Comment by vepicurean** [comment]
AFP and SMB file sharing protocol plugins. OMV Extras. I'm also using the Promax kernel.  I've enabled ZFS and that's the RAID volume my shares run on. 
That's about it that I added/configured after the initial OMV install. 

All worked fine (except for a strange email notification about an error each week about the SMARTdisk monitoring) til about two weeks ago. The install is basically brand new. 

All updates have been implemented.

**Comment by votdev** [comment]
Please check the syslog for more information. There are so many reasons why Monit can fail, so with this less information we can not do anything here.

**Comment by vepicurean** [comment]
What additional info do you need? This is not good. Is Openmediavault always this unstable just from doing a few updates?  It was fine a couple weeks ago. The reason I went with Openmediavault over BSD is I know and trust the Linux kernel and the Debian distribution. I run Debian Linux all over my systems and there's almost never an issue. I was expecting the same with OpenMediaVault, but it's obvious some of these plugins are not that stable.

**Comment by votdev** [comment]
Maybe starting Monit via systemd manually from CLI and watch the syslog may be a good starting point.

**Comment by vepicurean** [comment]
Normally I wouldn't mind taking more time to research the issue and help come up with a fix and share my findings.  

But I'm concerned this type of thing is going to be an on going issue with Openmediavault. For that reason I've avoided putting data on this new server for fear of losing it. I spent nearly a whole weekend researching this already, and though people have reported issues similar before, they are not describing exactly what I'm seeing. 

I think it's a major issue when the "Apply" function basically doesn't work and I've not done anything that could be considered non standard, core functionality in terms of configuration. 

It's a time issue. Before I invest the time (which is money expense to myself and others dependent on my time), usually in the Linux community others have seen the same issue before. Instead of me re-doing what others have already done, I was hoping someone in the Openmediavault community would have seen this before and have the fix.

**Comment by ryecoaaron** [comment]
> But I'm concerned this type of thing is going to be an on going issue with Openmediavault. 

Why?  I've never seen this issue on hundred of installs.  It could be a hardware or storage or ? issue that has nothing to do with OMV and can't be fixed with a code change.  When you use some thing free and have a problem that isn't common, you should expect to do the leg work.  If you find a problem that is caused by bad code, I'm sure it will be fixed if reproducible.

And remember, the web interface not working won't cause data loss.  In case of emergency, you could disable the omv services and manually configure samba/afp/zfs just like any other Linux system.

**Comment by vepicurean** [comment]
I hear you, but all you mention defeats the purpose of using OMV in the first place. If it's going to be this difficult to determine a fix to something like this, then might as well delete and rebuild the system using FreeNAS.....

Note: Nothing in this world is free (there's is always an exchange in the simplest interaction). All software/technology has a cost (time is money) whether the purchase price is 0 or not is often irrelevant.  That's been proven many times over. 

I'm even noticing this Monit port inaccessible error when doing the updates within OpenMediaVault. 

Here's someone/people talking about a similar issue (s) right here:  https://forum.openmediavault.org/index.php/Thread/12214-OMV-on-RPI2-Monit-error/

I reviewed all these and determined if this is going to keep happening and it's been around this long without a built in fix (maybe not exactly the same as what I'm seeing, but it's been seen before and seems prone to breaking even after a one time fix) then I'll ask the forum if there's a way to fix permanently before going a different route. That's all I'm doing.

**Comment by ryecoaaron** [comment]
It is only difficult because we aren't able to see everything about the system and you said you preferred not to do leg work. This is probably a failing OS drive that would cause ALL OSes including Windows and FreeNAS to have issues.

The monit port is only inaccessible while monit is being updated or restarted.  How would you avoid that??

The RPi2 post you mention is from 4 years when I first created the RPi images.  They were beta with a few issues and I am quite sure an old SD card was used.  So, this was a case of failing storage as well.  In case you think this is OMV-specific, VMware ESXi has this issue as well.

If OMV was broken this bad, hundreds if not thousands of users would be complaining and no one would use OMV.  I have been using OMV since the beginning on many systems and don't have these issues.  In order for OMV to be flexible like it is, sometimes a little Linux work is necessary.  If that isn't something you are willing to do, then a commercial NAS (FreeNAS is commercial as well) is probably what you need.

**Comment by votdev** [comment]
I think this is a Monit issue we can not do anything, it's how Monit works. There is a commit https://github.com/openmediavault/openmediavault/commit/970abaa4ee0a9c2f961680657da1f1d612b557ab that explains this behaviour.

```
Reduce monit delay from 30 to 5 seconds after the daemon is (re-)started. This is a delay until monit starts monitoring the configured services. During this time to CLI tool was not useable and results in an socket not available error. To allow the monitored services to be started before monit is starting monitoring, it will be started at least by systemd.
```

**Comment by vepicurean** [comment]
Thanks for the replies. The issue is not hardware related. If I didn't mention this before, my systems are the highest level of enterprise grade (yes, enterprise, above prosumer) performance. I just designed and build this system specifically to use with OpenMediaVault after doing my extensive research on OpenMediaVault vs Freenas. I burned/tested it beyond any conceivable field load for over a week. I do this for all my systems. This is not some hobbyist basement server using 10 year old throw away parts.  Hardware performance wise, it's perfect and I've verified that. 

That said, I expected some issues with OpenMediaVault software (the FreeNas people in the forums were generally right about this), but this issue with Monit and Openmediavault is a bit disappointing given what I've observed and what my research has uncovered. The system essentially doesn't work (nothing can be changed at least) while this Monit related condition exists. 

Before I go into the Openmediavault environment configuration, I want to confirm the resolution steps based on the "Reduce Monit delay from 30 to 5 seconds" post found here:

https://github.com/openmediavault/openmediavault/commit/970abaa4ee0a9c2f961680657da1f1d612b557ab

1) I go to the text file located at:

deb/openmediavault/usr/share/openmediavault/mkconf/monit

2) Open monit in a text edit and edit the the following line:

 OMV_MONIT_DELAY_SECONDS=${OMV_MONIT_DELAY_SECONDS:-"30"}

to read as the line:

OMV_MONIT_DELAY_SECONDS=${OMV_MONIT_DELAY_SECONDS:-"5"}

3) Save the file with edits and reboot the machine. 

Anything else that has to be done? 

Please confirm the resolution steps for me.  I try to avoid guessing.  

Then the obvious follow on question, if this is a known issue for a long time, and there's nothing that can be done with Monit within the Openmediavault application, other than this manual workaround to lower the delay variable setting from 30 seconds to 5 seconds, why could this not be set to 5 seconds from the beginning of the Openmediavault installation? 

At the very least, make it a clear part of the Openmediavault installation instructions?  So is not something that has to be researched and discovered when the Monit with Openmediavault issue occurs and Openmediavault is not working properly.

**Comment by votdev** [comment]
You need to edit ``/etc/default/openmediavault`` and add ``OMV_MONIT_DELAY_SECONDS=5``. After that execute ``omv-mkconf monit``.

> Then the obvious follow on question, if this is a known issue for a long time, and there's nothing that can be done with Monit within the Openmediavault application, other than this manual workaround to lower the delay variable setting from 30 seconds to 5 seconds, why could this not be set to 5 seconds from the beginning of the Openmediavault installation?

The specified delay tells monit how long to wait after the startup of the monit daemon to monitor the configured services.

We are using 30 seconds because otherwise monit will spam you with warning emails when a monitored service doesn't come up within the specified time. Maybe 5 seconds is enough when you have good hardware, but on a low powered ARM device 5 seconds are not enough because systemd will start monit and all other services in parallel. The con of this delay is that monit will not repsonsible during this time. Don't know why the developers of this tool decided to implement it that way.

**Comment by vepicurean** [comment]
I added the MONIT environment variable and rebooted. I noticed this messaging with starting MONIT.

![monit_startup_log_messages](https://user-images.githubusercontent.com/5864428/65835748-f4796600-e2b7-11e9-9eed-9833b840e93b.PNG)

**Comment by vepicurean** [comment]
Does anyone have the step by step instructions to resolve this Monit related error/issue on Openmediavault 100%? 

It seems to be a known issue, but no one has noted the exact steps to resolve the issue. Thanks for your help and time.

**Comment by vepicurean** [comment]
It's been three weeks. Not sure how this can be resolved when the issue is still occurring on my Openmediavault server and no one seems to have the steps to resolve. Describing the issue is not resolving it. I don't want to have to re-open the same issue again and lose the information in this string. I'm sure it is/will be important for others that will run into this same problem and are looking for the right fix.
